### PR TITLE
WebDriver: Implement `POST /session/{id}/window` endpoint

### PR DESCRIPTION
**Pull request by Tyrubias**
Implements another endpoint on #15551. Aside from the usual changes, this PR makes `WebContent::WebDriverConnection::get_property` a public static method rather than a private one since the `WebDriver` client code needs to extract properties from a `JsonValue` as well. Please let me know if there is a better practice I should follow with making this function commonly available.

### CODE REVIEW

File: Userland/Services/WebDriver/Session.cpp
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:163]: Missing spec link

File: Userland/Services/WebDriver/Session.h
Comment (linusg) [inline Userland/Services/WebDriver/Session.h:17]: This should not be included in the header, it's not needed here

**Review by linusg — CHANGES_REQUESTED** [review]
Nitpicks, deferring to @trflynn89 for an actual review

File: Userland/Services/WebDriver/Session.h
Comment (Tyrubias) [reply Userland/Services/WebDriver/Session.h:17]: I added this because I made `WebContent::WebDriverConnection::get_property` a public function in `WebContent/WebDriverConnection.h` since the window-switching code needs the functionality as well. Is there a better way to do this? I'm not that experienced with C++ best practices.

Comment (linusg) [reply Userland/Services/WebDriver/Session.h:17]: it's only used in the actual implementation, not the header, so should go into the cpp file :)

Comment (Tyrubias) [reply Userland/Services/WebDriver/Session.h:17]: Thanks! I'll fix that

File: Userland/Services/WebDriver/Session.cpp
Comment (TobyAsE) [inline Userland/Services/WebDriver/Session.cpp:171]: Instead of ´TODO´ I'd mark this and step 5 as ´FIXME´, all the other endpoints do that :^)

**Comment by trflynn89** [comment]
Hmm so I think ultimately, these Window commands are going to have to move into `WebContent::WebDriverConnection`. They will need to interact with Browser's / Ladybird's tabs; `WebDriverConnection` is the thing that has the ability to talk to those applications, so it probably makes more sense to move the existing Window commands over there first. 

As things are right now, this command won't really do anything, right? Because we currently only ever have one Window, you can't actually switch windows.

**Comment by Tyrubias** [comment]
I think I agree with what you're saying — it fits with our earlier discussion about implementing the new window endpoint. I'm a little fuzzy on the details, however. Doesn't `WebContent::WebDriverConnection` sit on the other end of the IPC connection? In order to implement these window commands we would still need these changes (and this Session object), but would also need to include an extra step of interacting with `WebContent` through IPC. Please correct me if I'm wrong — I'm very new to this.

**Comment by trflynn89** [comment]
Right, I'm suggesting that we entirely move `Session::get_window_handle`, `Session::close_window`, `Session::get_window_handles` over to WebContent. The way they're currently implemented isn't quite to spec (e.g. `Close the current top-level browsing context.` has a specific meaning we aren't fulfilling), so moving them to WebContent will clean that up. I think once that's done, we can implement things like the New Window command, by having WebContent talk to Browser/Ladybird to open a new tab.

**Comment by trflynn89** [comment]
Coincidental timing, but I actually ended up needing to do that move to make WebDriver work for Ladybird (#16066). So all of these Window commands will be in WebContent after that is merged.

**Comment by Tyrubias** [comment]
Ok I think I understand now. Once #16066 is merged I'll rebase and make these changes on top of that. Should be able to get "actual" window switching (and also new windows later) to work. Thank you for the help!

File: Userland/Services/WebContent/WebDriverConnection.cpp
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverConnection.cpp:489]: `get_property` does not return a reference to a string, so this should just be:
```c++
auto handle = TRY(get_property(payload, "handle"sv));
```

File: Userland/Services/WebContent/WebDriverConnection.cpp
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverConnection.cpp:492]: The way our dialogs are implemented, we can't switch tabs if a dialog is open. So we might as well implement this step and return:
```c++
if (m_page_client.page().has_pending_dialog())
    return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::UnexpectedAlertOpen, "A user dialog is open"sv);

```

File: Userland/Services/WebContent/WebDriverConnection.cpp
Comment (trflynn89) [inline Userland/Services/WebContent/WebDriverConnection.cpp:485]: Don't need the `AK::` prefix on `JsonValue`, and also use east-const please:
```c++
JsonValue const& payload
```

**Review by trflynn89 — CHANGES_REQUESTED** [review]

**Review by linusg — COMMENTED** [review]
LGTM, waiting for approval from @trflynn89

**Review by trflynn89 — APPROVED** [review]

**Pull request merged**
